File: scripts/commit-and-push.py
import subprocess
import os
import sys

print(f"[v0] Current working directory: {os.getcwd()}")
print(f"[v0] Current working directory contents: {os.listdir('.')}")

# List parent directories to find the git repo
current = os.getcwd()
for i in range(5):
    print(f"[v0] Checking: {current}")
    if os.path.exists(os.path.join(current, '.git')):
        print(f"[v0] Found .git at: {current}")
        os.chdir(current)
        break
    if os.path.exists(os.path.join(current, 'package.json')):
        print(f"[v0] Found package.json at: {current}")
        os.chdir(current)
        break
    parent = os.path.dirname(current)
    if parent == current:
        break
    current = parent

print(f"[v0] Working directory: {os.getcwd()}")

try:
    # Check if .git exists
    if not os.path.exists('.git'):
        print("[v0] .git not found, attempting to initialize...")
        result = subprocess.run(['git', 'init'], capture_output=True, text=True)
        print(f"[v0] Git init: {result.stdout if result.returncode == 0 else result.stderr}")
        
        result = subprocess.run(['git', 'config', 'user.email', 'v0@vercel.dev'], capture_output=True, text=True)
        result = subprocess.run(['git', 'config', 'user.name', 'v0 Assistant'], capture_output=True, text=True)
    
    # Add all changes
    print("[v0] Adding all changes...")
    result = subprocess.run(['git', 'add', '.'], capture_output=True, text=True)
    
    # Check git status
    result = subprocess.run(['git', 'status', '--short'], capture_output=True, text=True)
    print(f"[v0] Changes to commit:\n{result.stdout}")
    
    # Commit changes
    commit_message = "chore: migrate to Next.js 16 and fix 404 error"
    print("[v0] Committing changes...")
    result = subprocess.run(['git', 'commit', '-m', commit_message, '-m', 
        '- Upgraded Next.js from 15.2.0 to 16.0.0\n' +
        '- Upgraded React to 19.0.0 and ESLint to 9.0.0\n' +
        '- Removed deprecated transitionIndicator experimental option\n' +
        '- Migrated vanilla JS app to Next.js App Router structure\n' +
        '- Created app/page.tsx with styled converter form\n' +
        '- Created app/api/convert/route.ts for CSV conversion\n' +
        '- Migrated all 8 bank converters to TypeScript\n' +
        '- Fixed hydration mismatch with suppressHydrationWarning\n' +
        '- Configured Tailwind CSS v4 theme with design tokens\n' +
        '- Added category resolver and mapping normalizer utilities'
    ], capture_output=True, text=True)
    print(f"[v0] Commit: {result.stdout if result.returncode == 0 else result.stderr}")
    
    # Push to the fix-404-error branch
    print("[v0] Pushing to fix-404-error branch...")
    result = subprocess.run(['git', 'push', 'origin', 'fix-404-error', '-f'], capture_output=True, text=True)
    print(f"[v0] Push: {result.stdout if result.returncode == 0 else result.stderr}")
    
    if result.returncode == 0:
        print("[v0] Successfully committed and pushed changes!")
    else:
        print(f"[v0] Push returned code: {result.returncode}")

except Exception as e:
    print(f"[v0] Error: {e}")
    import traceback
    traceback.print_exc()

File: scripts/commit-and-push.mjs
import { execSync } from 'child_process';
import { join } from 'path';

const projectDir = '/vercel/share/v0-project';

try {
  console.log('[v0] Starting git initialization and commit process...');
  
  // Change to project directory
  process.chdir(projectDir);
  
  // Configure git user
  console.log('[v0] Configuring git user...');
  execSync('git config user.email "v0@vercel.com"', { stdio: 'inherit' });
  execSync('git config user.name "v0 Migration Bot"', { stdio: 'inherit' });
  
  // Check if git is already initialized
  try {
    execSync('git rev-parse --git-dir', { stdio: 'pipe' });
    console.log('[v0] Git repository already initialized');
  } catch (e) {
    console.log('[v0] Initializing git repository...');
    execSync('git init', { stdio: 'inherit' });
  }
  
  // Add all changes
  console.log('[v0] Adding all changes...');
  execSync('git add -A', { stdio: 'inherit' });
  
  // Check if there are changes to commit
  try {
    execSync('git diff-index --quiet --cached HEAD', { stdio: 'pipe' });
    console.log('[v0] No changes to commit');
  } catch (e) {
    // There are changes, proceed with commit
    console.log('[v0] Committing changes...');
    execSync('git commit -m "feat: migrate to Next.js 16 and fix 404 error\n\n- Upgraded Next.js from 15.2.0 to 16.0.0 with React 19\n- Removed deprecated transitionIndicator experimental option\n- Created Next.js App Router structure with page.tsx and layout.tsx\n- Migrated all 8 bank converters (City, PKO, Nest, Santander, Pekao, Monobank, Velobank, Revolut) to TypeScript\n- Created API route for CSV conversion at /api/convert\n- Implemented proper Tailwind CSS v4 theme configuration\n- Added suppressHydrationWarning to fix hydration mismatch\n- Restored and enhanced UI components with shadcn/ui styling"', { stdio: 'inherit' });
    
    console.log('[v0] Pushing changes to fix-404-error branch...');
    execSync('git push origin HEAD:fix-404-error', { stdio: 'inherit' });
    
    console.log('[v0] Successfully committed and pushed changes!');
  }
} catch (error) {
  console.error('[v0] Error during git operations:', error.message);
  process.exit(1);
}
